### PR TITLE
Hilt migration; migrate order screens

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of #3936, this migrates the rest of orders related screens to Hilt.

Please do not merge before #4138 

#### Testing
This PR touches lots of screens, I tried testing each screen during the migration, so small smoke testing of the orders should be enough 🤞 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt [hunk 1]
```diff
@@ -1,38 +1,22 @@
...
-import dagger.android.AndroidInjector
 import dagger.android.DispatchingAndroidInjector
-import dagger.android.HasAndroidInjector
-import dagger.android.support.AndroidSupportInjection
+import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
 
-class RefundItemsPickerDialog : NumberPickerDialog(), HasAndroidInjector {
+@AndroidEntryPoint
+class RefundItemsPickerDialog : NumberPickerDialog() {
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt:10]: ❓ Is it intentional that we remove "hasAndroidInjector" but keep the injection of `DispatchingAndroidInjector` in all dialogs? If it is, why do we need to keep the property?

**Review by malinajirka — APPROVED** [review]
Thanks @hichamboushaba ! I've left one question, but it looks good overall. I spend 15 minutes testing different flows on the order and related screens and everything worked as expected.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt:10]: No, it's not intentional, I just missed them, I'll remove them 👍

Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt:10]: 👍 Thanks

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/RefundItemsPickerDialog.kt:10]: Done in 8893c21

**Pull request merged**